lib/china_regions/helpers/form_helper.rb

```diff
@@ -59,16 +61,37 @@ def region_select(object_name, methods, options = {}, html_options = {})
 
       private
 
-      def get_choices(region_klass, method, preselected_choices, index)
-        choices = (index == 0 ? region_klass.where(nil).collect { |p| [ p.name, p.id ] } : [])
+      def prioritize_choices(priorities, method)
+        return if priorities.size == 0
+
+        if priorities[method]
+          sorting_map = {}
+          priorities[method].each_with_index do |value, i|
+            sorting_map[value.to_sym] = i
+          end
 
-        choices = preselected_choices[:cities] if preselected_choices[:cities] && method == :city
-        choices = preselected_choices[:districts] if preselected_choices[:districts] && method == :district
+          temp_choices = []
+          priority_choices = []
 
-        choices
+          @choices.each do |name, id|
+            if sorting_map[name.to_sym]
+              priority_choices[sorting_map[name.to_sym]] = [name, id]
+            else
+              temp_choices.push([name, id])
+            end
+          end
+          @choices = priority_choices.compact + temp_choices
+        end
       end
 
-      def set_preselected_choices(preselected_choices, options)
+      def get_choices(region_klass, method, index)
+        @choices = (index == 0 ? region_klass.where(nil).collect { |p| [ p.name, p.id ] } : [])
```

Comment:
This is very hard to understand. I would go with

```
choices = []
If index == 0
  choices = region_klass.where(nil).collect { |p| [ p.name, p.id ] }
end

if preselected_choices[:cities] && method == :city
  return preselected_choices[:cities]
elsif preselected_choices[:districts] && method == :district
  return preselected_choices[:districts]
end

if index == 0
  return region_klass.where(nil).collect { |p| [ p.name, p.id ] }
else
  return []
end
```

And even then, this just looks rough, because of the pre-selected choices part. Why as this broken up?